owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -161,21 +160,35 @@ class MainApp : Application() {
             }
         })
 
-        val newArchModule = module {
-            viewModel { (account: Account) ->
-                OCShareViewModel(androidContext(), account)
+        val viewModelModule = module {
+            viewModel { (filePath: String, account: Account) ->
+                OCShareViewModel(
```

Comment:
why do you need context??

Reply:
It's a strong dependency I need to create ownCloud Clients and execute network operations. I already fixed it using dependency injection in coming PRs.

Reply:
ok